### PR TITLE
Dynamically set jvm

### PR DESCRIPTION
**Pull request by danielba32**

### CODE REVIEW

File: opensearch-operator/pkg/builders/cluster.go [hunk 1]
```diff
@@ -162,9 +165,37 @@ func NewSTSForNodePool(
...
+		reG := regexp.MustCompile(`G`)
+		// Convert Gb to Mb
+		if reG.MatchString(memString) {
+			memInt = memInt * 1024
+		}
+
+		// Calculate jvm memory limit by cluster role
+		if isDataNode {
+			newMemory := int(math.Round(float64(memInt) * 0.5))
+			newMemoryStr := strconv.Itoa(newMemory) + "M"
```
Comment (idanl21) [inline opensearch-operator/pkg/builders/cluster.go:189]: If the user is configure JVM with Gi or Mi is it also work ?

**Review by idanl21 — COMMENTED** [review]
Hey @danielba32 
Looks good, but we have to add some explantion on that on our user gudie and tests that created cluster with 2g of meme and checks if the calculation worked for data nodes and non-data node

File: opensearch-operator/pkg/builders/cluster.go (hunk 1 above)
Comment (danielba32) [reply opensearch-operator/pkg/builders/cluster.go:189]: if user set pod resorces with Gi it convert it to mb in any way the jvm gonna be in Mb when the jvm not specified

**Comment by swoehrl-mw** [comment]
Hi @danielba32. Are you still working on this PR or should we close it?

**Comment by swoehrl-mw** [comment]
Superseded by #558. Closing this PR.

**Pull request closed without merging**